atramhasis/scripts/sitemap_generator.py

```diff
@@ -0,0 +1,163 @@
+        raise
+    finally:
+        session.close()
+
+
+def write_element_to_xml(filename, sitemap_dir, element):
+    tree = ElementTree.ElementTree(element)
+    file_path_name = os.path.join(sitemap_dir, filename)
+    tree.write(file_path_name, encoding='utf-8', xml_declaration=True)
+
+
+def create_sitemaps(settings, limit_per_deel, directory, env):
+    base_url = settings.get("atramhasis.url")
+    schemes_url = "{}{}".format(base_url, '/conceptschemes/{}')
+    concepts_url = "{}{}".format(schemes_url, '/c/{}')
+
+    request = env['request']
+
+    if hasattr(request, 'skos_registry') and request.skos_registry is not None:
+        skos_registry = request.skos_registry
+    else:
+        raise SkosRegistryNotFoundException()  # pragma: no cover
+
+    scheme_urls = list()
+    concept_urls = list()
+    for p in skos_registry.get_providers():
+        if any([not_shown in p.get_metadata()['subject']
+                for not_shown in ['external', 'hidden']]):
+            continue
+        scheme_id = p.get_metadata()['id']
+        scheme_urls.append(schemes_url.format(scheme_id))
+        concept_urls.extend([concepts_url.format(scheme_id, x['id'])
+                             for x in p.get_all()])
+
+    create_deel_sitemaps(scheme_urls, limit_per_deel, directory, 'conceptschemes')
+    create_deel_sitemaps(concept_urls, limit_per_deel, directory, 'concepts')
+
+    create_index_sitemap(base_url, directory)
+
+
+def create_deel_sitemaps(objecturls, limit_per_deel, sitemap_dir, name):
+    """
+    Sitemaps have a maximum amount of items. This method creates partial
+    sitemaps with up to `limit_per_deel` items per file.
+    """
+    log.info("Beginning creation of sitemaps...")
+    urlset = ElementTree.Element(
+        "urlset", xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
+    )
+    sitemap_counter = 1
+
+    for counter, objecturl in enumerate(objecturls, 1):
+        url = ElementTree.SubElement(urlset, "url")
+        ElementTree.SubElement(url, "loc").text = objecturl
+
+        if counter % limit_per_deel == 0:
+            filename = '{}_sitemap_deel_{}.xml'.format(name, sitemap_counter)
+            log.info("Processed %s conceptschemes, writing %s", counter, filename)
+            write_element_to_xml(filename, sitemap_dir, urlset)
+            sitemap_counter += 1
+            urlset = ElementTree.Element(
+                "urlset", xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
+            )
+    if len(urlset):
+        filename = '{}_sitemap_deel_{}.xml'.format(name, sitemap_counter)
+        write_element_to_xml(filename, sitemap_dir, urlset)
+    log.info("All {} sitemaps created.".format(name))
+
+
+def create_index_sitemap(base_url, directory):
+    """Loop over all the created sitemaps, and create an index file."""
+    log.info("Beginning creation of the final sitemap index...")
+    list_sitemaps = [f for f in listdir(
+        directory) if isfile(os.path.join(directory, f))
```

Comment:
Technisch gezien is dit niet fout. Maar ik vind de indentatie een beetje confusing.